spec/controllers/V1/tasks_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe V1::TasksController, :type => :controller do
+
+  let!(:project) { create(:project) }
+
+  describe "PUT update_task_state" do
+    let!(:task) { create(:task, project_id: project.id) }
+    let!(:update_params) do { project_id: project.id, task_id: task.id, state: "in-progress" } end
```

Comment:
Also, I might put the `update_params` in a context `when a task is updated to in-progress` like you did down below when updating to `done`. That way it's a little more clear.